timemachine/cpp/src/harmonic_bond.cu

```diff
@@ -62,11 +62,11 @@ template <typename RealType> HarmonicBond<RealType>::~HarmonicBond() {
 
 template <typename RealType>
 void HarmonicBond<RealType>::execute_device(
-    const int N,
```

Comment:
Why are some names of these removed?

Reply:
This was to satisfy `Wunused-parameter` (which is enabled as part of `-Wextra`). Agreed that this seems suboptimal. The warning seems useful to have in some cases, but maybe not worth this.

Reply:
Oof - this does remove useful self-documentation information. Hm... I personally like the old style better.

Reply:
I added `-Wno-unused-parameter` and reverted these changes in 36549c3